WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/StatsPage.java

```diff
@@ -0,0 +1,179 @@
+                onView(withText("Posts and Pages"))
+        );
+
+        return this;
+    }
+
+    public StatsPage scrollToPosts() {
+        scrollToCard("Posts and Pages");
+        return this;
+    }
+
+    public StatsPage scrollToReferrers() {
+        scrollToCard("Referrers");
+        return this;
+    }
+
+    public StatsPage scrollToClicks() {
+        scrollToCard("Clicks");
+        return this;
+    }
+
+    public StatsPage scrollToAuthors() {
+        scrollToCard("Authors");
+        return this;
+    }
+
+    public StatsPage scrollToCountries() {
+        scrollToCard("Countries");
+        return this;
+    }
+
+    public StatsPage scrollToVideos() {
+        scrollToCard("Videos");
+        return this;
+    }
+
+    public StatsPage scrollToFileDownloads() {
+        scrollToCard("File downloads");
+        return this;
+    }
+
+    public StatsPage assertVisits(StatsVisitsData visitsData) {
+            ViewInteraction cardStructure = onView(allOf(
+                    withId(R.id.stats_block_list),
+                    hasDescendant(allOf(
+                            withText("Views"),
+                            hasSibling(withText(visitsData.getViews()))
+                            )
+                    ),
+                    hasDescendant(allOf(
+                            withText("Visitors"),
+                            hasSibling(withText(visitsData.getVisitors()))
+                            )
+                    ),
+                    hasDescendant(allOf(
+                            withText("Likes"),
+                            hasSibling(withText(visitsData.getLikes()))
+                            )
+                    ),
+                    hasDescendant(allOf(
+                            withText("Comments"),
+                            hasSibling(withText(visitsData.getComments()))
+                            )
+                    )
+            ));
+
+            cardStructure.check(matches(isCompletelyDisplayed()));
+            return this;
+    }
+
+    public void assertKeyValuePairs(String cardHeader, List<StatsKeyValueData> list) {
+        for (StatsKeyValueData item : list) {
+            // Element with ID = stats_block_list
+            // |--Has child with text: e.g. "Posts and Pages"
+            // |--Has descendant that both:
+            //    |- Has text: post.title
+            //    |- Has a sibling with post.views (which means they're shown on same row):
+            ViewInteraction cardStructure = onView(allOf(
+                    withId(R.id.stats_block_list),
+                    hasDescendant(withText(cardHeader)),
+                    hasDescendant(allOf(
+                            withText(item.getKey()),
+                            hasSibling(withText(item.getValue()))
+                            )
+                    )
+            ));
+
+            cardStructure.check(matches(isCompletelyDisplayed()));
+        }
+    }
```

Comment:
This method is called by all (except one) asserting methods. It receives a stat card header name as an argument (e.g. "Posts and Pages"), and a list of key-value pairs as another argument:

<img width="706" alt="Screenshot 2022-01-21 at 12 48 12" src="https://user-images.githubusercontent.com/73365754/150514472-ca465940-89bc-463e-afa6-9dbdaf443593.png">

What is then does is making sure that these pairs really belong to a card with certain header and each pair is really a pair:

<img width="421" alt="Screenshot 2022-01-20 at 22 30 37" src="https://user-images.githubusercontent.com/73365754/150513905-d5c49dca-83e5-4586-85e4-e40208b321b8.png">

For the record, this is the structure of the card in layout inspector:

<img width="390" alt="Screenshot 2022-01-20 at 22 36 19" src="https://user-images.githubusercontent.com/73365754/150514602-773b2e1d-6a18-42ae-8fb2-48f4767e4ffb.png">